support-workers/src/main/scala/com/gu/zuora/subscriptionBuilders/DigitalSubscriptionPurchaseBuilder.scala

```diff
@@ -28,15 +28,16 @@ class DigitalSubscriptionPurchaseBuilder(
     requestId: UUID,
     environment: TouchPointEnvironment,
     maybeGiftCode: Option[GeneratedGiftCode],
+    maybeDeliveryDate: Option[LocalDate],
   )(implicit ec: ExecutionContext): Either[PromoError, SubscriptionData] = {
 
     val productRatePlanId = validateRatePlan(digitalRatePlan(digitalPack, environment), digitalPack.describe)
 
-    val (contractAcceptanceDelay, autoRenew, initialTerm, maybeRedemptionCode) =
-      (digitalPack.readerType, maybeGiftCode) match {
-        case (Gift, Some(giftCode)) => (0, false, GiftCodeValidator.expirationTimeInMonths + 1, Some(giftCode))
-        case (Gift, _) | (_, Some(_)) => throw new RuntimeException("coding error - possible unredeemable sub")
-        case _ => (config.defaultFreeTrialPeriod + config.paymentGracePeriod, true, 12, None)
+    val (contractAcceptanceDelay, autoRenew, initialTerm, maybeRedemptionCode, deliveryDate) =
+      (digitalPack.readerType, maybeGiftCode, maybeDeliveryDate) match {
+        case (Gift, Some(giftCode), Some(deliveryDate)) => (0, false, GiftCodeValidator.expirationTimeInMonths + 1, Some(giftCode), Some(deliveryDate))
+        case (Gift, _, _) | (_, Some(_), _) | (Gift, _, None) => throw new RuntimeException("coding error - possible unredeemable sub")
+        case _ => (config.defaultFreeTrialPeriod + config.paymentGracePeriod, true, 12, None, None)
```

Comment:
now it will blow up if we get a Gift but we don't have a delivery date to put into zuora, otherwise we might not notice.  But it does mean that if we break that, all gift checkouts will fail, rather than going through with missing data.
If we think that's too harsh but still want some checking, we could make it a metric.
Alternatively, once we finish refactoring the state types, this is the sort of thing that would be checked by the type system anyway.

Reply:
I think that's fine